server/routes/ratings.js

```diff
@@ -13,71 +14,60 @@ router.get('/', function (req, res) {
   });
 });
 
-/* GET rating by id */
-router.get('/:id', function (req, res) {
-  models.Rating.findById(req.params.id).then( function (rating) {
-    res.send(rating.toJson());
-  });
-});
-
-/* UPDATE user's rating for snippet*/
-router.put('/:id', function (req, res) {
-  models.Rating.findById(req.params.id).then( function (rating) {
-    rating.value = req.body.value;
-    rating.save({ validate: false, logging: true}).then(function () {
-      res.status(200).send('ok');
-    }).catch(function () {
-      res.status(422).send('error');
-    });
-  }).catch(function () {
-    res.status(422).send('error');
-  });
-});
-
 /* POST new rating */
 router.post('/', function (req, res) {
   var userId = req.user.get('id');
   var snippetId = req.body.SnippetId;
-  var sumRatings = 0.0;
-  models.Rating.findOne({ where : { SnippetId: snippetId, UserId: userId } }).then( function (rating) {
-    if (rating) {
-      rating.value = req.body.value;
-      rating.save({ validate: false, logging: true}).then(function () {
-        models.Rating.sum('value', { where : { SnippetId: snippetId } }).then( function (sum) {
-          sumRatings = sum;
-          models.Rating.count({ where : { SnippetId: snippetId } }).then( function (c){
-            var average = 0;
-            if(c>0){
-              average = (sumRatings/c);
-            }
-            res.status(200).send({rating: rating, avg: average.toFixed(2)});
+  var attributes = {
+    value: req.body.value,
+    UserId: userId,
+    SnippetId: snippetId
+  };
+  models.sequelize.transaction(function (t){
+    return models.Rating.findOne({ where : { SnippetId: snippetId, UserId: userId }, transaction: t }).then( function (rating) {
+      if (rating) {
+        return rating.update(attributes, {transaction: t}).then( function () {
+          return models.Rating.aggregate('value', 'avg', { where : { SnippetId : snippetId }, dataType: 'float', transaction: t }).then( function (avg) {
+            return models.Snippet.findById(snippetId, {transaction: t}).then( function (s) {
```

Comment:
1. You can call `update` providing `{ where: { id: x }}` and it will return instance, there's no need for using explicit `findById` just to update and it works fine
BUT
in this case we should additionaly check whether `currentUser` is owner of rated snippet ;)